odoo/exceptions.py

```diff
@@ -13,24 +13,31 @@
 import logging
 from inspect import currentframe
 from .tools.func import frame_codeinfo
+from .tools.translate import _
 
 _logger = logging.getLogger(__name__)
 
 
 # kept for backward compatibility
 class except_orm(Exception):
-    def __init__(self, name, value=None):
+    def __init__(self, name, value=None, action=None):
         if type(self) == except_orm:
             caller = frame_codeinfo(currentframe(), 1)
             _logger.warn('except_orm is deprecated. Please use specific exceptions like UserError or AccessError. Caller: %s:%s', *caller)
         self.name = name
         self.value = value
         self.args = (name, value)
 
+        # only send the action to the web client if the user has the right permissions
+        if action and action.get('visible', True):
+            self.action = action
+        else:
+            self.action = False
```

Comment:
The permission is defined where the action is defined.
This code processes something that says: "*I made some effort to compute something that is useless.*"